crates/bevy_ecs/src/schedule/ambiguity_detection.rs

```diff
@@ -0,0 +1,814 @@
+use crate::component::ComponentId;
+use crate::schedule::{AmbiguityDetection, SystemContainer, SystemStage};
+use crate::world::World;
+
+use bevy_utils::get_short_name;
+use bevy_utils::tracing::{error, warn};
+use fixedbitset::FixedBitSet;
+use std::hash::Hash;
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+/// Systems that access the same Component or Resource within the same stage
+/// risk an ambiguous order that could result in logic bugs, unless they have an
+/// explicit execution ordering constraint between them.
+///
+/// This occurs because, in the absence of explicit constraints, systems are executed in
+/// an unstable, arbitrary order within each stage that may vary between runs and frames.
+///
+/// Some ambiguities reported by the ambiguity checker may be warranted (to allow two systems to run
+/// without blocking each other) or spurious, as the exact combination of archetypes used may
+/// prevent them from ever conflicting during actual gameplay.
+/// You can resolve the warnings produced by the ambiguity checker by adding `.before` or `.after`
+/// to one of the conflicting systems referencing the other system to force a specific ordering.
+/// Alternatively, if you're confident the error is a false positive (and you don't need true determinism),
+/// you can explicitly ignore it using the `.ambiguous_with` method.
+///
+/// Note that the checker reports each pairwise ambiguity:
+/// typically, these can be resolved with fewer constraints than the number of ambiguities listed
+/// as transitive orderings will resolve ambiguities (e.g. A before B before C will resolve an ambiguity between A and C).
+/// It's generally more productive to pause and reflect on the underlying structure of your logic,
+/// rather than blindly resolving one warning at a time.
+///
+/// By default, the value of this resource is set to `Warn`.
+pub enum ExecutionOrderAmbiguities {
```

Comment:
```suggestion
/// Systems that access the same Component or Resource within the same stage
/// risk an ambiguous order that could result in logic bugs, unless they have an
/// explicit execution ordering constraint between them.
///
/// This occurs because, in the absence of explicit constraints, systems are executed in
/// an unstable, arbitrary order within each stage that may vary between runs and frames.
///
/// Some ambiguities reported by the ambiguity checker may be warranted (to allow two systems to run
/// without blocking each other) or spurious, as the exact combination of archetypes used may
/// prevent them from ever conflicting during actual gameplay.
/// You can resolve the warnings produced by the ambiguity checker by adding `.before` or `.after`
/// to one of the conflicting systems referencing the other system to force a specific ordering.
/// Alternatively, if you're confident the error is a false positive (and you don't need true determinism),
/// you can explicitly ignore it using the `.ambiguous_with` method.
///
/// Note that the checker reports each pairwise ambiguity:
/// typically, these can be resolved with fewer constraints than the number of ambiguities listed
/// as transitive orderings will resolve ambiguities (e.g. A before B before C will resolve an ambiguity between A and C).
/// It's generally more productive to pause and reflect on the underlying structure of your logic,
/// rather than blindly resolving one warning at a time.
///
/// By default, the value of this resource is set to `Warn`.
#[derive(Copy, Clone, Debug, PartialEq, Eq)]
pub enum ExecutionOrderAmbiguities {
```